test/unit/specs/components/Personalization/createComponent.spec.js

```diff
@@ -10,94 +10,247 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-import { defer } from "../../../../../src/utils";
 import createComponent from "../../../../../src/components/Personalization/createComponent";
-import createConfig from "../../../../../src/core/config/createConfig";
-import {
-  PAGE_WIDE_SCOPE_DECISIONS,
-  PAGE_WIDE_SCOPE_DECISIONS_WITHOUT_DOM_ACTION_SCHEMA_ITEMS,
-  SCOPES_FOO1_FOO2_DECISIONS
-} from "./responsesMock/eventResponses";
+import * as SCHEMAS from "../../../../../src/constants/schemas";
+import { values } from "../../../../../src/utils";
 
-describe("Personalization", () => {
-  const config = createConfig({ prehidingStyle: "" });
-  const logger = {
-    log() {},
-    warn() {}
-  };
-  const eventManager = {
-    createEvent() {},
-    sendEvent() {}
-  };
+const ALL_SCHEMAS = values(SCHEMAS);
 
+describe("Personalization", () => {
+  let logger;
+  let config;
+  let onResponseHandler;
+  let onClickHandler;
+  let hideContainers;
+  let showContainers;
+  let hasScopes;
+  let isAuthoringModeEnabled;
+  let getDecisionScopes;
+  let mergeQuery;
+  let createQueryDetails;
   let event;
-  let response;
 
   beforeEach(() => {
     event = jasmine.createSpyObj("event", ["mergeQuery"]);
-    response = jasmine.createSpyObj("response", ["getPayloadsByType"]);
+
+    logger = {
+      info: jasmine.createSpy(),
+      warn: jasmine.createSpy()
+    };
+    isAuthoringModeEnabled = jasmine.createSpy().and.returnValue(false);
+    config = jasmine.createSpy();
+    onResponseHandler = jasmine.createSpy();
+    onClickHandler = jasmine.createSpy();
+    hideContainers = jasmine.createSpy();
+    showContainers = jasmine.createSpy();
+    hasScopes = jasmine.createSpy();
+    getDecisionScopes = jasmine.createSpy();
+    mergeQuery = jasmine.createSpy();
+    createQueryDetails = jasmine.createSpy();
   });
 
-  it("should return an array of decisions in lifecycle::onResponse for provided decisions decisionScopes", () => {
-    const decisionScopes = ["Foo1", "Foo3"];
+  it("shouldn't do anything since authoringMode is enabled", () => {
+    isAuthoringModeEnabled.and.returnValue(true);
+    const personalizationComponent = createComponent({
+      config,
+      logger,
+      onResponseHandler,
+      onClickHandler,
+      hideContainers,
+      showContainers,
+      hasScopes,
+      isAuthoringModeEnabled,
+      getDecisionScopes,
+      mergeQuery,
+      createQueryDetails
+    });
```

Comment:
If you want, you can extract this out to a `build` function like was done here: https://github.com/adobe/alloy/blob/b7fc7d8e68eeb7cec68201c9205ef83584a638bb/test/unit/specs/components/Privacy/createComponent.spec.js#L37-L46

Completely your choice though.

Reply:
ok, thank you